### DISCUSSION TITLE
Error starting Peacock

### DISCUSSION THREAD

**Discussion by RuSmith96** [idaholab/moose#21606] [Q&A Tools] [answered]
Hi, 

I have recently installed MOOSE on Ubuntu but I am unable to start up Peacock. I am getting this error message ->

Error starting peacock: Could not import vtk

You need to use the miniconda module. Run:
	module load miniconda

I had already installed miniconda for the installation of MOOSE and I have also installed the additional libraries/packages from the manual_miniconda section in installation. But I am still getting this error. I am assuming I have missed something? The MOOSE installation passed or skipped with no fails so I thought this would be okay. 

Any help would be great!
Thanks in advance.

**Comment by GiudGiud** [comment, marked as answer]
Hello

mamba install vtk should be all you need to do to fix this

Guillaume

**Reply by milljm** [reply]
Actually there is a new package available that installs everything necessary to launch Peacock. The only caveat, is that you must install this package into a new environment, and remember to activate this environment whenever you want to use Peacock:

```bash
mamba activate base     # get back to the default environment
mamba create -n peacock moose-peacock
mamba activate peacock
<run peacock>
```

You can run any number of different environments, so long as you run then in different terminal windows.

While currently installing moose-peacock into your running moose environment (the environment that you created during the 'Getting Started' steps: `mamba create -n moose moose-libmesh moose-tools`) _might work now_. It will not work in the future. So save yourself the headache, and separate these environments now.

**Reply by GiudGiud** [reply]
I would do mamba install moose-libmesh moose-tools moose-peacock since I would want to run the moose apps from peacock?

Either way, this is great, will simplify installing peacock for sure!

**Reply by milljm** [reply]
You can run Peacock in a different environment than what was used to build moose-based applications.

The reason we want to instruct users to separate environments running `moose-libmesh` and `moose-peacock` now, is because the two environments will eventually differ to the point where `mamba install moose-libmesh moose-peacock` will cause conflicts.

The reason: PyQT is not being built using every compiler being made available on conda-forge. At some point, when we move to Clang 13, PyQt may _not_. And thus; an impossible solve situation during an install. Best to teach folks now, to keep these separate.

**Reply by Ofinal** [reply]
Hello, i used the ' `mamba create -n  peacock python=3.10`' command to create a separate environment for '`peacock`'  on CentOS 7, but then my screen showed "nothing provides requested moose -peacock" when i  typed the '`mamba install moose-peacock`' into the terminal. I wanna know how to install peacock on a separate environment created by mamba.
Thank you for your replying!

Best Regards
